src/libcore/mem.rs

```diff
@@ -65,6 +76,24 @@ pub use intrinsics::transmute;
 /// // Leak an I/O object, never closing the file
 /// let file = File::open("foo.txt").unwrap();
 /// mem::forget(file);
+///
+/// // The swap function uses forget
+/// fn swap<T>(x: &mut T, y: &mut T) {
+///     unsafe {
+///         // Give ourselves some scratch space to work with
+///         let mut t: T = uninitialized();
+///
+///         // Perform the swap, `&mut` pointers never alias
+///         ptr::copy_nonoverlapping(&*x, &mut t, 1);
```

Comment:
I think that this code example will need to import `std::ptr` to compile, as well as change the call to `uninitialized` to `mem::uninitialized`. You can be sure to run the doc tests via `make check-stage1-doc-crate-core`.